packages/terser/src/terser_minified.js

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env node
+/**
+ * @fileoverview wraps the terser CLI to support minifying a directory
+ * Terser doesn't support it; see https://github.com/terser/terser/issues/75
+ * TODO: maybe we should generalize this to a package which would be useful outside
+ *       bazel; however we would have to support the full terser CLI and not make
+ *       assumptions about how the argv looks.
+ */
+const fs = require('fs');
+const path = require('path');
+const child_process = require('child_process');
+const os = require('os')
+
+// Run Bazel with --define=VERBOSE_LOGS=1 to enable this logging
+const VERBOSE_LOGS = !!process.env['VERBOSE_LOGS'];
+
+function log_verbose(...m) {
+  if (VERBOSE_LOGS) console.error('[terser/index.js]', ...m);
+}
+
+function log_error(...m) {
+  console.error('[terser/index.js]', ...m);
+}
+
+/**
+ * @param {string[]} residualArgs
+ * @param {string} output
+ */
+function processSourceMapArg(residualArgs, output) {
+  // copy args so we don't accidently mutate and process the same set of args twice
+  const argsCopy = [...residualArgs];
+  const sourceMapIndex = argsCopy.findIndex(a => a === '--source-map');
+  if (sourceMapIndex != -1) {
+    // the options for the source map arg is the next one
+    // if it changes in terser_minified.bzl this needs to be updated
```

Comment:
we should consider reintroducing minimist for option parsing at the very least to organize which flags we read and why.

Reply:
i'd appciate having a lib for handling args, doing it manually can get error prone.
Considering almost everthing we handle is CLIs maybe this is a good idea.
Or at very least implement our own standward for CLI args parsing if we dont want to pull in a full lib